clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security/AmazonClientProvider.java

```diff
@@ -274,6 +275,45 @@ public AmazonCloudWatch getCloudWatch(NetflixAmazonCredentials amazonCredentials
     }
   }
 
+  private boolean isNTLMProxy(NetflixAmazonCredentials amazonCredentials){
+
+    boolean isNTLMProxy = false;
+
+    AmazonCredentials.AWSProxy proxy = amazonCredentials.getAWSProxy();
+
+    if(proxy != null){
+
+      if(proxy.getProxyHost() != null && proxy.getProxyPort() != null && proxy.getProxyDomain() != null && proxy.getProxyWorkstation() != null )
+        isNTLMProxy = true;
+
+    }
+
+    return isNTLMProxy;
+  }
+
+  private boolean isProxyConfigMode(NetflixAmazonCredentials amazonCredentials){
+
+    boolean isProxy = false;
+
+    AmazonCredentials.AWSProxy proxy = amazonCredentials.getAWSProxy();
+
+    if(proxy != null){
+
+      if(proxy.getProxyHost() != null && proxy.getProxyPort() != null)
+
+        isProxy = true;
+        try{
+          Integer.parseInt(proxy.getProxyPort());
+        }catch(NumberFormatException nfe){
+           isProxy = false;
+        }
+
+
+    }
+
+    return isProxy;
```

Comment:
`return proxy != null && proxy.isValid()'

Might be simpler if these methods were actually on `AWSProxy`.